File: generated-sources/java-vertx/mojang-api/src/main/java/org/openapitools/server/api/model/OrderStatistic.java
package org.openapitools.server.api.model;

import java.util.Objects;

/**
 * Valid options to query the order statistics
 **/
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Valid options to query the order statistics
 */
public enum OrderStatistic {
  
  ITEM_SOLD_MINECRAFT("item_sold_minecraft"),
  
  PREPAID_CARD_REDEEMED_MINECRAFT("prepaid_card_redeemed_minecraft"),
  
  ITEM_SOLD_COBALT("item_sold_cobalt"),
  
  ITEM_SOLD_SCROLLS("item_sold_scrolls"),
  
  PREPAID_CARD_REDEEMED_COBALT("prepaid_card_redeemed_cobalt"),
  
  ITEM_SOLD_DUNGEONS("item_sold_dungeons");

  private String value;

  OrderStatistic(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static OrderStatistic fromValue(String text) {
    for (OrderStatistic b : OrderStatistic.values()) {
      if (String.valueOf(b.value).equals(text)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + text + "'");
  }
}

File: generated-sources/kotlin-spring/mojang-api/src/main/kotlin/org/openapitools/model/OrderStatistic.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonValue
import javax.validation.Valid
import javax.validation.constraints.*

/**
* Valid options to query the order statistics
* Values: itemSoldMinecraft,prepaidCardRedeemedMinecraft,itemSoldCobalt,itemSoldScrolls,prepaidCardRedeemedCobalt,itemSoldDungeons
*/
enum class OrderStatistic(val value: String) {

    itemSoldMinecraft("item_sold_minecraft"),

    prepaidCardRedeemedMinecraft("prepaid_card_redeemed_minecraft"),

    itemSoldCobalt("item_sold_cobalt"),

    itemSoldScrolls("item_sold_scrolls"),

    prepaidCardRedeemedCobalt("prepaid_card_redeemed_cobalt"),

    itemSoldDungeons("item_sold_dungeons");

}

